leonardnjura/github-users
# #156185242 Setup app tests

#### What does this PR do?
- Manages dependency versions with variables defined in project build.gradle

#### Description of Task to be completed?
- Define dependencies in one place
- Refer to them using variable names

#### How should this be manually tested?
* Clone this repo: 
`$ git clone -b ft-testing-app-156185242 https://github.com/leonardnjura/github-users`
* Open cloned directory `github-users` in Android Studio
* Build with Gradle
* Connect an emulator or physical device
* Navigate to
`app/../test` and `app/../androidTest` 
* Run unit and instrumented tests using by right-clicking on IDE item or in a terminal using:
`$ ./gradlew test ` and `./gradlew connectedAndroidTest` 
* Expect passing tests
* To get a test report run
`./gradlew connectedAndroidTest`
* Open this Jacoco report in your browser after the tests
`app/build/reports/jacoco/jacocoTestReport/html/index.html`


#### What are the relevant pivotal tracker stories?
[#156185242](https://www.pivotaltracker.com/story/show/156185242)
[#157354404](https://www.pivotaltracker.com/story/show/157354404)
[#156185411](https://www.pivotaltracker.com/story/show/156185411)


#### Screenshots
![ezgif com-video-to-gif (4)](https://user-images.githubusercontent.com/39657549/54823563-4c0b9900-4cb9-11e9-8e28-28484537d3b5.gif)

![ezgif com-video-to-gif (3)](https://user-images.githubusercontent.com/39657549/54823564-4c0b9900-4cb9-11e9-9538-aa250e87149c.gif)

![Screenshot 2019-03-22 at 16 34 40](https://user-images.githubusercontent.com/39657549/54826273-7b71d400-4cc0-11e9-977e-eb580ab42cd9.png)
